tests/rules/no-invalid-at-rules.test.js

```diff
@@ -293,5 +295,206 @@ ruleTester.run("no-invalid-at-rules", rule, {
 				},
 			],
 		},
+		{
+			code: '@CHARSET "UTF-8";',
+			output: '@charset "UTF-8";',
+			errors: [
+				{
+					messageId: "unknownAtRule",
+					data: { name: "CHARSET" },
+					line: 1,
+					column: 1,
+					endLine: 1,
+					endColumn: 9,
+				},
+			],
+		},
+		{
+			code: '@CharSet "UTF-8";',
+			output: '@charset "UTF-8";',
+			errors: [
+				{
+					messageId: "unknownAtRule",
+					data: { name: "CharSet" },
+					line: 1,
+					column: 1,
+					endLine: 1,
+					endColumn: 9,
+				},
+			],
+		},
+		{
+			code: "@charset",
+			errors: [
+				{
+					messageId: "missingPrelude",
+					data: { name: "charset" },
+					line: 1,
+					column: 1,
+					endLine: 1,
+					endColumn: 9,
+				},
+			],
+		},
+		{
+			code: "@charset;",
+			errors: [
+				{
+					messageId: "missingPrelude",
+					data: { name: "charset" },
+					line: 1,
+					column: 1,
+					endLine: 1,
+					endColumn: 9,
+				},
+			],
+		},
+		{
+			code: "@charset ;",
+			errors: [
+				{
+					messageId: "missingPrelude",
+					data: { name: "charset" },
+					line: 1,
+					column: 1,
+					endLine: 1,
+					endColumn: 9,
+				},
+			],
+		},
+		{
+			code: "@charset 'UTF-8';",
+			output: '@charset "UTF-8";',
+			errors: [
+				{
+					messageId: "invalidPrelude",
+					data: {
+						name: "charset",
+						prelude: "'UTF-8'",
+						expected: "<string>",
+					},
+					line: 1,
+					column: 10,
+					endLine: 1,
+					endColumn: 17,
+				},
+			],
+		},
```

Comment:
In the case of a syntax error in `@charset`, the error message `invalidPrelude` seems incorrect, such as in this test, the message would be:

```
Invalid prelude ''UTF-8'' found for at-rule '@charset'. Expected '<string>'.
```

Here it says, `<string>` is expected to be used, which is already used here, but with wrong quotes.

And in the case of  `@charset "UTF-8"`, where a semicolon is missing, it has the same error message.

What do you think about adding an error message only to report syntax errors?

Reply:
Yes, but i feel this error message seems a bit descriptive; instead, we can just suggest the correct syntax, like for - 
`@charset "UTF-8"` message can be something like this:

```
Invalid @charset syntax. Expected '@charset "UTF-8";'
```

Please feel free to choose the words that you find more accurate here.